tests/unit/lms/db/_bulk_action_test.py

```diff
@@ -55,9 +55,15 @@ def test_upsert(self, db_session):
             ).only()
         )
 
+    def test_upsert_does_nothing_if_given_an_empty_list_of_values(self, db_session):
+        assert BulkAction(db_session).upsert(self.TableWithBulkUpsert, []) == []
+
     def test_it_fails_with_missing_config(self, db_session):
         with pytest.raises(AttributeError):
-            BulkAction(db_session).upsert("object_without_config", [])
+            BulkAction(db_session).upsert(
+                "object_without_config",
+                [{"id": 1, "name": "name", "other": "other"}],
+            )
```

Comment:
This test was previously passing an empty list to `upsert()` bit wasn't triggering the crash because the test table didn't have any non-nullable columns. I changed it to not pass an empty list so that it doesn't run into the special case